ddtrace/contrib/aiopg/connection.py

```diff
@@ -15,21 +15,19 @@ class AIOTracedCursor(wrapt.ObjectProxy):
     def __init__(self, cursor, pin):
         super(AIOTracedCursor, self).__init__(cursor)
         pin.onto(self)
-        name = pin.app or 'sql'
-        self._datadog_name = '%s.query' % name
 
     @asyncio.coroutine
-    def _trace_method(self, method, resource, extra_tags, *args, **kwargs):
+    def _trace_method(self, method, query, extra_tags, *args, **kwargs):
         pin = Pin.get_from(self)
         if not pin or not pin.enabled():
             result = yield from method(*args, **kwargs)  # noqa: E999
             return result
         service = pin.service
 
-        with pin.tracer.trace(self._datadog_name, service=service,
-                              resource=resource) as s:
+        name = (pin.app or 'sql') + "." + method.__name__
+        with pin.tracer.trace(name, service=service) as s:
             s.span_type = sql.TYPE
-            s.set_tag(sql.QUERY, resource)
+            s.set_tag(sql.QUERY, query or self.query.decode('utf-8'))
```

Comment:
The sql query must be set as a `resource` otherwise we can't offer correct stats. Also, the `sql.query` tag must not be set otherwise the obfuscation is skipped in the Agent, and even if PostgreSQL uses prepared statement (so it's a kind of obfuscated), it's better to rely in our obfuscation tool.

Reply:
removed